azext_edge/edge/commands_edge.py

```diff
@@ -229,16 +229,16 @@ def init(
         mq_insecure=mq_insecure,
         target_name=target_name,
         keyvault_resource_id=keyvault_resource_id,
-        keyvault_sat_secret_name=str(keyvault_sat_secret_name),
+        keyvault_spc_secret_name=str(keyvault_spc_secret_name),
         disable_secret_rotation=disable_secret_rotation,
         rotation_poll_interval=str(rotation_poll_interval),
         service_principal_app_id=service_principal_app_id,
         service_principal_object_id=service_principal_object_id,
         service_principal_secret=service_principal_secret,
-        service_principal_secret_valid_days=service_principal_secret_valid_days,
+        service_principal_secret_valid_days=int(service_principal_secret_valid_days),
```

Comment:
Reason being with Az CLI the default int/string/simple type assignment sets the type as `knack.validators.Default[Str|Int]` etc instead of the native Python type.